fuzz-test-suite/dateparserfuzzer.cpp

```diff
@@ -0,0 +1,20 @@
+#include <ql/time/asx.hpp>
+#include <ql/time/date.hpp>
+#include <ql/time/ecb.hpp>
+#include <ql/time/imm.hpp>
+#include <ql/time/timeunit.hpp>
+#include <ql/utilities/dataparsers.hpp>
+#include <string>
```

Comment:
Is this header required for Clang’s built-in fuzzer? If not, then a preprocessor macro should be used to check for the fuzzer, to avoid adding the external header dependency when not needed.

Reply:
This header is shipped with clang/llvm it is not technically required but it is a nice helper. It's not used for detection rather it takes a string of raw bytes and turns them into structured data, like std::string I've used it on line 11. It also handles seperating data between calls so in this case it'll ensure that a different string is passed to each parser. A fuzzer is essentially a super basic AI that tries to maximise code coverage by randomly mutating the input. So making sure subsets of the input go to different code paths is important to fuzzing performance.

Reply:
I can remove this dependency if it's important but I'll end up reinventing the wheel a little.

Reply:
If the header is shipped with clang then probably fine to leave it in there. I might add the following just to make it clear to people who happen to try out a different compiler:

```c++
#ifndef __clang__
#pragma message(“fuzzer headers are available from clang, other compilers have not been tested”)
#endif
```

Reply:
Noted I'll add that in now

Reply:
Done